CONTRIBUTING.md

```diff
@@ -0,0 +1,84 @@
+
+After this ADR is reviewed and approved, the feature can be taken out of the experimental toggle and will be considered an official part of the project, and stable for production use
+
+¹ This flag is already scoped to be deprecated, so that we can provide individual feature toggle for experimental features.
+² This is not implemented yet, but it is already planned.
+³ A base model for the ADR will be provided soon.
+
+## Contribution Guidelines
+
+### Cachi2's Ethos
+Whenever adding a new feature to Cachi2, it is important to keep these fundamental aspects in mind:
+
+1. Very accurate reporting of the prefetched dependencies
+
+    Cachi2's primary goal is to prefetch content and enable hermetic builds. But hermetic builds are only useful if they end up providing a more accurate SBOM than a non-hermetic build would. Cachi2 strives to download only what's explicitly declared in a project's source code, and accurately report it in the resulting SBOM.
+
+2. Avoid arbitrary code execution
+
+    Some package manager implementations rely on third-party tools to gather data or even to fetch the dependencies. This brings the risk of arbitrary code execution, which opens the door for random content to be part of the prefetched content. This is undermines the accuracy of the SBOM, and must be avoided at all costs.
+
+3. Always perform checksum validation
+
+    The content provided to the build will only be safe if all the downloaded packages had their checksums verified. In case a mismatch is found, the full request can be failed, since the prefetched content is tainted and is potentially malicious. There are two types of checksums: server-provided and user-provided. Cachi2 prefers the latter, but does not require it. Everytime that a dependency did not have a user-provided checksum verified, the resulting SBOM should state so. All dependencies need at least one checksum to be validated.
+
+4. Favor reproducibilty
+
+    Always use fully resolved lockfiles or similar input files to determine what content needs to be download for a specific project (e.g. npm's `package-lock.json`, a `pip-compiled` generated `requirements.txt`, etc). Resolving the dependencies during the prefetch will prevent its behavior from being deterministic, or in other words, the same repository and the same commit hash should always result in identical prefetch results.
+
+
+### Coding Guidelines
+- Write clear and informative commit messages.
+- Make use of the description part of the commit message to provide further context.
+- Split changes in multiple commits, in a way that each commit 
```

Comment:
...that each commit addresses a clear and concise problem.

Reply:
Fixed